e2e/tests/transfer/authz_test.go

```diff
@@ -233,107 +237,111 @@ func (suite *AuthzTransferTestSuite) TestAuthz_InvalidTransferAuthorizations() {
+			chainBIBCToken := testsuite.GetIBCToken(chainADenom, chainAChannels.Counterparty.PortID, chainAChannels.Counterparty.ChannelID)
+			actualBalance, err := s.QueryBalance(ctx, chainB, receiverWalletAddress, chainBIBCToken.IBCDenom())
 
-			suite.Require().NoError(err)
-			suite.Require().Equal(int64(0), actualBalance.Int64())
+			s.Require().NoError(err)
+			s.Require().Equal(int64(0), actualBalance.Int64())
 		})
 
 		t.Run("granter grant spend limit unchanged", func(t *testing.T) {
-			grantAuths, err := suite.QueryGranterGrants(ctx, chainA, granterAddress)
+			grantAuths, err := s.QueryGranterGrants(ctx, chainA, granterAddress)
 
-			suite.Require().NoError(err)
-			suite.Require().Len(grantAuths, 1)
+			s.Require().NoError(err)
+			s.Require().Len(grantAuths, 1)
 			grantAuthorization := grantAuths[0]
 
-			transferAuth := suite.extractTransferAuthorizationFromGrantAuthorization(grantAuthorization)
+			transferAuth := s.extractTransferAuthorizationFromGrantAuthorization(grantAuthorization)
 			expectedSpendLimit := sdk.NewCoins(sdk.NewCoin(chainADenom, sdkmath.NewInt(spendLimit)))
-			suite.Require().Equal(expectedSpendLimit, transferAuth.Allocations[0].SpendLimit)
+			s.Require().Equal(expectedSpendLimit, transferAuth.Allocations[0].SpendLimit)
 		})
 	})
 
 	t.Run("send funds to invalid address", func(t *testing.T) {
-		invalidWallet := suite.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+		invalidWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
 		invalidWalletAddress := invalidWallet.FormattedAddress()
 
 		t.Run("broadcast MsgExec for ibc MsgTransfer", func(t *testing.T) {
 			transferMsg := transfertypes.MsgTransfer{
-				SourcePort:    channelA.PortID,
-				SourceChannel: channelA.ChannelID,
+				SourcePort:    chainAChannels.PortID,
+				SourceChannel: chainAChannels.ChannelID,
 				Token:         sdk.Coin{Denom: chainADenom, Amount: sdkmath.NewInt(spendLimit)},
 				Sender:        granterAddress,
 				Receiver:      invalidWalletAddress,
-				TimeoutHeight: suite.GetTimeoutHeight(ctx, chainB),
+				TimeoutHeight: s.GetTimeoutHeight(ctx, chainB),
 			}
 
 			protoAny, err := codectypes.NewAnyWithValue(&transferMsg)
-			suite.Require().NoError(err)
+			s.Require().NoError(err)
 
 			msgExec := &authz.MsgExec{
 				Grantee: granteeAddress,
 				Msgs:    []*codectypes.Any{protoAny},
 			}
 
-			resp := suite.BroadcastMessages(context.TODO(), chainA, granteeWallet, msgExec)
+			resp := s.BroadcastMessages(context.TODO(), chainA, granteeWallet, msgExec)
 			if testvalues.IbcErrorsFeatureReleases.IsSupported(chainAVersion) {
-				suite.AssertTxFailure(resp, ibcerrors.ErrInvalidAddress)
+				s.AssertTxFailure(resp, ibcerrors.ErrInvalidAddress)
 			} else {
-				suite.AssertTxFailure(resp, sdkerrors.ErrInvalidAddress)
+				s.AssertTxFailure(resp, sdkerrors.ErrInvalidAddress)
 			}
 		})
 	})
+
+	t.Run("stop relayer", func(t *testing.T) {
+		s.StopRelayer(ctx, relayer)
+	})
```

Comment:
do we need this addition?

Reply:
we had this problem if we didn't StopRelayer `tried to start relayer again without stopping first`